### PR TITLE
improve servlet error capture

### LINKED ISSUE #711: Capture exception from spring boot @ExceptionHandler
context: https://discuss.elastic.co/t/apm-dont-capture-exception-from-exceptionhandler/187530

### PR DESCRIPTION
**Pull request by kananindzya**
### Checklist

closes elastic/apm-agent-java#711 
- [x] Implement code
- [x] Add tests
- [ ] ~Update documentation~
- [ ] Update [CHANGELOG.md](CHANGELOG.md)
- [ ] ~Update [supported-technologies.asciidoc](docs/supported-technologies.asciidoc)~
- [ ] ~Added an API method or config option? Document in which version this will be introduced.~

### CODE REVIEW

**Comment by kananindzya** [comment]
@eyalkoren , @felixbarny please check PR

**Comment by kananindzya** [comment]
> LGTM. Please add a changelog entry for this +1

added

File: apm-agent-plugins/apm-spring-webmvc-plugin/src/test/java/co/elastic/apm/agent/spring/webmvc/exception/AbstractExceptionHandlerInstrumentationTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    @Before
+    @BeforeEach
+    public void setup() {
+        this.mockMvc =
+            MockMvcBuilders.webAppContextSetup(wac).build();
+    }
+
+    protected void assertExceptionCapture(int spanCount, int errorCount, Class exceptionClazz, MockHttpServletResponse response, int statusCode, String responseContent, String exceptionMessage) throws UnsupportedEncodingException {
```
Comment (eyalkoren) [inline apm-agent-plugins/apm-spring-webmvc-plugin/src/test/java/co/elastic/apm/agent/spring/webmvc/exception/AbstractExceptionHandlerInstrumentationTest.java]: No need for the `errorCount` argument - now we are in the proper state where all tests produce exactly one error capture.
`spanCount` is always 0, so you can remove this argument as well.

File: apm-agent-plugins/apm-spring-webmvc-plugin/src/test/java/co/elastic/apm/agent/spring/webmvc/exception/AbstractExceptionHandlerInstrumentationTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    @BeforeClass
+    @BeforeAll
+    public static void setUpAll() {
+        reporter = new MockReporter();
+        tracer = new ElasticApmTracerBuilder()
+            .configurationRegistry(SpyConfiguration.createSpyConfig())
+            .reporter(reporter)
+            .build();
+        ElasticApmAgent.initInstrumentation(tracer, ByteBuddyAgent.install(),
+            Arrays.asList(new ServletInstrumentation(tracer), new ExceptionHandlerInstrumentation()));
```
Comment (eyalkoren) [inline apm-agent-plugins/apm-spring-webmvc-plugin/src/test/java/co/elastic/apm/agent/spring/webmvc/exception/AbstractExceptionHandlerInstrumentationTest.java]: If you do:
```suggestion
            Arrays.asList(new ServletInstrumentation(tracer)));
```
instead, does the relevant test fail now?

File: apm-agent-plugins/apm-spring-webmvc-plugin/src/test/java/co/elastic/apm/agent/spring/webmvc/exception/AbstractExceptionHandlerInstrumentationTest.java (hunk 1 above)
Comment (kananindzya) [reply apm-agent-plugins/apm-spring-webmvc-plugin/src/test/java/co/elastic/apm/agent/spring/webmvc/exception/AbstractExceptionHandlerInstrumentationTest.java]: removed

File: apm-agent-plugins/apm-spring-webmvc-plugin/src/test/java/co/elastic/apm/agent/spring/webmvc/exception/AbstractExceptionHandlerInstrumentationTest.java (hunk 2 above)
Comment (kananindzya) [reply apm-agent-plugins/apm-spring-webmvc-plugin/src/test/java/co/elastic/apm/agent/spring/webmvc/exception/AbstractExceptionHandlerInstrumentationTest.java]: with instrumentation:
image
without:
image

Comment (kananindzya) [reply apm-agent-plugins/apm-spring-webmvc-plugin/src/test/java/co/elastic/apm/agent/spring/webmvc/exception/AbstractExceptionHandlerInstrumentationTest.java]: Yes, it's fail now

Comment (eyalkoren) [reply apm-agent-plugins/apm-spring-webmvc-plugin/src/test/java/co/elastic/apm/agent/spring/webmvc/exception/AbstractExceptionHandlerInstrumentationTest.java]: Great, then it is properly tested 🙂 
Thanks!

**Pull request merged**